ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -0,0 +1,306 @@
+import re
+
+# NOTE (iurygregory): most of the sensor readings come in the ipmi format
+# each type of sensor consider a different range of values that aren't integers
+# (eg: 0h, 2eh), 0h will be published as 0 and the other values as 1, this way
+# we will be able to create prometheus alerts.
+# Documentation: https://www.intel.com/content/www/us/en/servers/ipmi/
+# ipmi-second-gen-interface-spec-v2-rev1-1.html
+
+
+def add_prometheus_type(name, metric_type):
+    return '# TYPE %s %s' % (name, metric_type)
+
+
+def metric_names(payload, prefix, sufix, extract_unit=False,
+                 special_label=None):
+    metric_dic = {}
+    for entry in payload:
+        if special_label == 'fan':
+            e = re.sub(r'[\d].*$', '', entry.lower())
```

Comment:
Maybe it's the github PR UI, but I don't see any tests that demonstrate the expected output for all of the parsing in this function. Am I missing it?

Reply:
You're not missing, I'm going to add some tests for it.